ivy/functional/backends/paddle/elementwise.py

```diff
@@ -53,10 +52,9 @@ def bitwise_xor(
     return paddle.bitwise_xor(x1, x2)
 
 
+@with_supported_dtypes({"2.5.1 and below": ("float",)}, backend_version)
 def expm1(x: paddle.Tensor, /, *, out: Optional[paddle.Tensor] = None) -> paddle.Tensor:
-    if x.dtype in [paddle.float16, paddle.float32, paddle.float64]:
-        return paddle.expm1(x)
-    return paddle_backend.subtract(paddle_backend.exp(x), 1.0).astype(x.dtype)
```

Comment:
could you please test this out if the `astype` in the output is necessary? Probably we can consider the other dtypes as supported too.

Reply:
there's a problem with this casting if the input `dtype `is an `int` because this `paddle_backend.subtract(paddle_backend.exp(x), 1.0)` will return a `float `and the `.astype(x.dtype)` will cause a loss in precision 
![image](https://github.com/unifyai/ivy/assets/100947451/4bf7c8e5-c352-4933-8d20-d8d8141324a9)
but if the input is float we get a correct answer:
![image](https://github.com/unifyai/ivy/assets/100947451/e4fc8231-12cc-4a55-a4b7-6338b6dd773a)

Reply:
But that's the expected precision anyway right if the output dtype is expected to be `int16` because of the input dtypes that were passed?

Reply:
I'm not sure if  want to allow this kind of behavior, If you want me to keep these dtypes I'll do it ?

Reply:
So just to understand your point, we're basically trying to compute `expm1` with an `int` to start with in the example you shared right? What's the output dtype with the other backends? The array api says that both the inputs and outputs should have a `float` dtype. In the ivy tests we're only testing this function with [float and complex](https://github.com/unifyai/ivy/blob/c26a5c80fd919a241babbe9e69cc5bf9145ed5bd/ivy_tests/test_ivy/test_functional/test_core/test_elementwise.py#L771) numbers which is of course not correct.
Do you think it's better for us to mark `int` as unsupported and cast them to `float` when the casting modes are used? Or do you think we should pass `int` to the function and return a `float` output? Thanks!

Reply:
yes, I'm trying to pass an `int` type as an input.
tensorflow does not support `int` types for this function.
jax and numpy and torch cast the output to `float32`:
```python
import jax.numpy as jnp
x = jnp.array([1, 1, 1], dtype=jnp.int16)
jnp.expm1(x)
Out[11]: Array([1.7182817, 1.7182817, 1.7182817], dtype=float32)
```
```python
x = np.array([1, 1, 1], dtype=np.int16)
np.expm1(x)
Out[16]: array([1.7182817, 1.7182817, 1.7182817], dtype=float32)
```
```python
import torch
x = torch.tensor([1, 1, 1], dtype=torch.int16)
torch.expm1(x)
Out[20]: tensor([1.7183, 1.7183, 1.7183])
_.dtype
Out[21]: torch.float32
```
so the only problem is with tensorflow and paddle(precision problem) which can be resolved by the dtype decorators for tensorflow. And paddle I prefer to raise error or mark them to unsupported instead of the precision loss (ideally we would ask the client what he really wants), because I think the expected return type is `float` even when the input is an `int`. For the other backends that does support an `int` type, I'd say to mark the int as supported since they behave correctly in my view.
what do you think ??

Reply:
got it, thanks for clarifying! So you mean that `float32` should be the expected `dtype` of the output when an `int` is passed to `expm1`? Sounds good to me, as long as we're consistent and the behaviour is intuitive 👍